template/package.json

```diff
@@ -32,10 +32,11 @@
     "fix:js": "npm run lint:js -- --fix",
     "fix:md": "npm run lint:md -- -o",
     "fix:deps": "npm run lint:deps -- --update",
+    "prepare": "node checkPackageScope.js",
```

Comment:
where and how is this run?

since this is already enforced this through the template itself, wouldn't the right place to do this be in a CI environment?

Reply:
@ahmadnassri That was my original approach, however I then realized that we already have what we need in the project to parse `package.json`, so it's easier to just have a script for it. It's a `prepare` script, which according to the docs runs `both BEFORE the package is packed and published, on local npm install without any arguments, and when installing git dependencies`. I thought this makes sense because people shouldn't even get to push something with a non-scoped package name.

I can just make it a custom script and have it run as a separate job or in the lint job in CI. Or I can do it entirely in CI, but then I gotta install whatever I need to parse the json in there.

Now that I gave more context, what are your thoughts? This is minimum effort and no added work/delays in the CI container.

Reply:
It's not enforced through the template. Folks can just initialize with the correct name format but then update `package.json` and omit `@telus`. Nothing stops them from doing that and releasing right now.

Reply:
ah, I see, the `prepare` npm-scripts explainer clarified ...

in that case I would recommend using `prepublish` because `prepend` runs on every `npm install` and that seems wasteful